paddle/fluid/platform/mkldnn_reuse.h

```diff
@@ -542,6 +562,73 @@ class ConvMKLDNNTemplateHandler : public MKLDNNHandler {
                                scale_data, mask);
   }
 
+  mkldnn::primitive_attr CreatePostOps(bool fuse_relu,
+                                       bool fuse_residual_conn = false) const {
+    mkldnn::primitive_attr conv_attr;
+    mkldnn::post_ops post_operations;
+    // Fusion with Elementwise layer relies on adding a sum post-operation with
+    // the scale parameter. It is assumed that when fuse_residual_connection is
+    // true, the output tensor contains the data coming from residual
+    // connection. The result of this post_op is:
+    // Output = scale * Output + Conv_Out.
+    if (fuse_residual_conn) {
+      post_operations.append_sum(1.0f);
+    }
+    // Fusion with ReLU layer is executed through the PostOps feature. Create a
+    // PostOps object and configure it to execute an eltwise relu operation.
+    if (fuse_relu) {
+      constexpr float scale = 1.0f;
+      constexpr float negative_slope = 0.0f;
+      constexpr float placeholder = 0.0f;
+      post_operations.append_eltwise(scale, mkldnn::algorithm::eltwise_relu,
+                                     negative_slope, placeholder);
+    }
+    conv_attr.set_post_ops(post_operations);
+    return conv_attr;
+  }
+
+  std::shared_ptr<typename forward_t::primitive_desc>
+  AcquireConvolutionPrimitiveDescriptor(
+      const mkldnn::memory::desc& src, const mkldnn::memory::desc& weights,
+      boost::optional<const mkldnn::memory::desc&> bias,
+      const mkldnn::memory::desc& dst, const std::vector<int>& strides,
+      const std::vector<int>& paddings, const mkldnn::engine& engine,
+      const bool fuse_relu, const bool fuse_residual_conn,
+      mkldnn::prop_kind fwd_prop_kind) {
+    const std::string key_conv_pd = key_ + "@conv_pd";
+
+    auto conv_pd = std::static_pointer_cast<typename forward_t::primitive_desc>(
+        dev_ctx_.GetBlob(key_conv_pd));
+
+    if (conv_pd == nullptr) {
+      mkldnn::memory::dims stride_dims = strides;
+      mkldnn::memory::dims padding_dims = paddings;
+
+      auto conv_desc =
+          bias ? typename forward_t::desc(fwd_prop_kind, mode<forward_t>::T,
```

Comment:
And it is just for conv now, just use what is needed seems clearer.

Reply:
@tensor-tang AcquireConvolutionPrimitiveDescriptor is used for both conv and conv_transpose.

Reply:
All reference just: https://github.com/PaddlePaddle/Paddle/pull/16667/files#diff-a60a81d3c97ec5a7cf7cf3d7e1aa9adbR608 and https://github.com/PaddlePaddle/Paddle/pull/16667/files#diff-a60a81d3c97ec5a7cf7cf3d7e1aa9adbR612 right?

seems no need add `mode` for these.

Reply:
@tensor-tang I do not quite understand. AcquireConvolutionPrimitiveDescriptor is used twice: for conv and conv_transpose. So for conv , template paramter is "convolution_forward" and later on "mode" does set algorithm based on that to "convolution_direct" . For conv_transpose template parameter is "deconvolution_forward" and later on "mode" maps this deconvolution_forward into "deconvolution_direct". So in my opinion it is useful at that scenario.

Reply:
OK.
Then the name should be adapted, `mode` is not a good name for what is mean. 
`algorithm` should be better, or other names you suggest.

Reply:
I introduced change : "mode" -> "convolutional_algorithm"  . Hopefully it is more descriptive that way.